compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3628,6 +3755,86 @@ private BLangExpression createVarCheckCondition(BLangMatchPattern matchPattern,
         }
     }
 
+    private BLangExpression createVarCheckCondition(BLangBindingPattern bindingPattern, BLangSimpleVarRef varRef) {
+        NodeKind bindingPatternKind = bindingPattern.getKind();
+        Location pos = bindingPattern.pos;
+        switch (bindingPatternKind) {
+            case WILDCARD_BINDING_PATTERN:
+                return createConditionForWildCardBindingPattern(true, pos);
+            case CAPTURE_BINDING_PATTERN:
+                return createConditionForCaptureBindingPattern((BLangCaptureBindingPattern) bindingPattern, varRef,
+                        pos);
+            case MAPPING_BINDING_PATTERN:
+                return createVarCheckConditionForMappingBindingPattern((BLangMappingBindingPattern) bindingPattern,
+                        varRef);
+            default:
+                // If some patterns are not implemented, those should be detected before this phase
+                // TODO : Remove this after all patterns are implemented
+                return null;
+        }
+    }
+
+    private BLangExpression createVarCheckConditionForMappingBindingPattern(BLangMappingBindingPattern
```

Comment:
The logic in this method too long for readability. Shall we break this method into few for better readability? 
I know that existing methods in Desugar class doesn't follow it, but we can start adding it for any new additions IMO.

Reply:
Yes, fixed.